src/Notes/WC_Admin_Notes_Install_Jetpack_And_WooCommerce_Services_Plugin.php

```diff
@@ -0,0 +1,121 @@
+ *
+ * Adds a note to the merchant's inbox prompting them to install the Jetpack
+ * Plugin after it fails to install during WooCommerce setup.
+ *
+ * @package WooCommerce Admin
+ */
+
+namespace Automattic\WooCommerce\Admin\Notes;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * WC_Admin_Notes_Install_Jetpack_And_WooCommerce_Services_Plugin
+ */
+class WC_Admin_Notes_Install_Jetpack_And_WooCommerce_Services_Plugin {
+	const NOTE_NAME = 'wc-admin-install-jetpack-plugin';
+
+	/**
+	 * Constructor.
+	 */
+	public function __construct() {
+		add_action( 'woocommerce_note_action_install-jetpack-and-woocommerce-services-plugins', array( $this, 'install_jetpack_and_woocommerce_services_plugins' ) );
+
+		$notes_filters = new WC_Admin_Notes_Filters();
+		add_action( 'activate_jetpack/jetpack.php', array( $notes_filters, 'action_jetpack_woocommerce_services_activation_note' ) );
+		add_action( 'activate_woocommerce-services/woocommerce-services.php', array( $notes_filters, 'action_jetpack_woocommerce_services_activation_note' ) );
+	}
+
+	/**
+	 * Possibly add the Install Jetpack note.
+	 *
+	 * @param string $slug The slug of the plugin being installed.
+	 */
+	public static function possibly_add_install_jetpack_and_woocommerce_services_note( $slug ) {
+		// Exit early if we're not installing the Jetpack or the WooCommerce Services plugins.
+		if ( 'jetpack' !== $slug && 'woocommerce-services' !== $slug ) {
+			return;
+		}
+
+		$data_store = \WC_Data_Store::load( 'admin-note' );
+
+		// Exit early if there is already a note to install Jetpack.
+		$note_ids = $data_store->get_notes_with_name( self::NOTE_NAME );
+		if ( ! empty( $note_ids ) ) {
+			return;
+		}
+
+		$content = __( 'We noticed that there was a problem during the Jetpack and WooCommerce Services install. Please try again and enjoy all the advantages of having the plugins connected to your store! Sorry for the inconvenience. The "Jetpack" and "WooCommerce Services" plugins will be installed & activated for free.', 'woocommerce-admin' );
+
+		$note = new WC_Admin_Note();
+		$note->set_title( __( 'Uh oh...There was a problem during the Jetpack and WooCommerce Services install. Please try again.', 'woocommerce-admin' ) );
+		$note->set_content( $content );
+		$note->set_content_data( (object) array() );
+		$note->set_type( WC_ADMIN_Note::E_WC_ADMIN_NOTE_INFORMATIONAL );
+		$note->set_icon( 'mail' );
+		$note->set_name( self::NOTE_NAME );
+		$note->set_source( 'woocommerce-admin' );
+		$note->add_action(
+			'install-jetpack-and-woocommerce-services-plugins',
+			__( 'Install plugins', 'woocommerce-admin' ),
+			false,
+			WC_Admin_Note::E_WC_ADMIN_NOTE_ACTIONED,
+			true
+		);
+
+		$note->save();
+	}
+
+	/**
+	 * Action the Install Jetpack notes, if any exists.
+	 */
+	public static function action_install_jetpack_and_woocommerce_services_note() {
+		$data_store = \WC_Data_Store::load( 'admin-note' );
+		$note_ids   = $data_store->get_notes_with_name( self::NOTE_NAME );
+
```

Comment:
This was filled with sharks - `PluginsHelper::get_active_plugin_slugs()` didn't return the just-activated plugin in the `activated_{$plugin}` action so I had to do some hacking - but it's done in 5ae961760ef52a4933f576ea4ae9c24082299f0f

Reply:
Is that perhaps because [`activate_{$plugin}`](https://github.com/WordPress/WordPress/blob/master/wp-admin/includes/plugin.php#L690) occurs before the plugin has been activated?

Maybe hooking into [`activated_plugin`](https://github.com/WordPress/WordPress/blob/master/wp-admin/includes/plugin.php#L717) would allow for the removal of the `action_note_wcs()` and `action_note_jp()` intermediate functions.

Reply:
Awesome, I missed that hook somehow. Fixed in 8bc90853f43ab79f8f980fc6a4acfeeb619999e3 which simplifies things quite a bit.